packages/wrangler/src/dev.tsx

```diff
@@ -469,13 +487,14 @@ export async function devHandler(args: ArgumentsCamelCase<DevArgs>) {
 					usageModel={config.usage_model}
 					bindings={bindings}
 					crons={config.triggers.crons}
+					isApi={args.isApi}
+					onReady={args.onReady}
 				/>
 			);
 		}
-		const { waitUntilExit, rerender } = render(
-			await getDevReactElement(config)
-		);
-		await waitUntilExit();
+		const devReactElement = render(await getDevReactElement(config));
+		rerender = devReactElement.rerender;
+		return { devReactElement, watcher };
```

Comment:
How about we return our own `waitForExit` function here that does the `watcher.close()` for us?
E.g.

```ts
async function waitForExit() {
  try {
    await devReactElement.waitUntilExit();
  } finally {
    watcher?.close();
  }
}
return { waitForExit };
```

Then in the caller we don't need to know about the watcher:

```ts
export async function devHandler(args: ArgumentsCamelCase<DevArgs>) {
		const { waitForExit } = await startDev(args);
		await waitUntilExit();
}
```